feat(docs-infra): generate the sitemap for angular.dev

josephperrott opened the pull request:
We automatically generate the sitemap for the angular.dev deployment and include it in the assets pushed to firebase.

josephperrott committed:
feat(docs-infra): generate the sitemap for angular.dev

We automatically generate the sitemap for the angular.dev deployment and include it in the assets pushed
to firebase.

Co-authored-by: Matthieu Riegler <kyro38@gmail.com>

JeanMeche commented:
we ran into a failure https://github.com/angular/angular/actions/runs/19079289664/job/54503568959 on the 20.3.x branch. We'll need to fix it before we merge the others branches

josephperrott committed:
fix(docs-infra): properly expose the prerendered routes for sitemap generation

Properly expose the prerendered routes for sitemap generation rather than only copying the browser directory

thePunderWoman commented:
@josephperrott This appears to have conflicts.
